CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+#[[
+This source file is part of the Swift open source project
+
+Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See http://swift.org/LICENSE.txt for license information
+See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+]]
+
+cmake_minimum_required(VERSION 3.26...3.29)
+project(SwiftBuild
+  LANGUAGES C CXX Swift)
+
+set(CMAKE_C_VISIBILITY hidden)
+set(CMAKE_CXX_VISIBILITY hidden)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
+set(CMAKE_CXX_EXTENSIONS NO)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN YES)
+
+option(SwiftBuild_USE_LLBUILD_FRAMEWORK "Use LLBuild Framework" NO)
+
+add_compile_options("$<$<COMPILE_LANGUAGE:Swift>:SHELL:-package-name SwiftBuild>"
+                    "$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<VERSION_LESS:$<TARGET_PROPERTY:Swift_LANGUAGE_VERSION>,6>>:SHELL:-enable-upcoming-feature ConciseMagicFile>"
+                    "$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<VERSION_LESS:$<TARGET_PROPERTY:Swift_LANGUAGE_VERSION>,6>>:SHELL:-enable-upcoming-feature DeprecateApplicationMain>"
+                    "$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<VERSION_LESS:$<TARGET_PROPERTY:Swift_LANGUAGE_VERSION>,6>>:SHELL:-enable-upcoming-feature DisableOutwardActorInference>"
+                    "$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<VERSION_LESS:$<TARGET_PROPERTY:Swift_LANGUAGE_VERSION>,6>>:SHELL:-enable-upcoming-feature ForwardTrailingClosures>"
+                    "$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<VERSION_LESS:$<TARGET_PROPERTY:Swift_LANGUAGE_VERSION>,6>>:SHELL:-enable-upcoming-feature GlobalConcurrency>"
+                    "$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<VERSION_LESS:$<TARGET_PROPERTY:Swift_LANGUAGE_VERSION>,6>>:SHELL:-enable-upcoming-feature ImplicitOpenExistentials>"
+                    "$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<VERSION_LESS:$<TARGET_PROPERTY:Swift_LANGUAGE_VERSION>,6>>:SHELL:-enable-upcoming-feature ImplicitOpenExistentialsImportObjcForwardDeclarations>"
+                    "$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<VERSION_LESS:$<TARGET_PROPERTY:Swift_LANGUAGE_VERSION>,6>>:SHELL:-enable-upcoming-feature InferSendableFromCaptures>"
+                    "$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<VERSION_LESS:$<TARGET_PROPERTY:Swift_LANGUAGE_VERSION>,6>>:SHELL:-enable-upcoming-feature IsolatedDefaultValues>"
+                    # rdar://127809703
+                    # "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-upcoming-feature RegionBasedIsolation>"
+                    "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-upcoming-feature ExistentialAny>"
+                    "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-upcoming-feature InternalImportsByDefault>")
+
+find_package(ArgumentParser)
+find_package(LLBuild)
+find_package(SwiftDriver)
+find_package(SwiftSystem)
```

Comment:
One thing to consider here is that the sources are currently setup to allow using CryptoKit and System from Darwin SDKs in place of swift-crypto and swift-system. Currently, the OS copies are preferred when they exist.

Reply:
Not sure what you have in mind. None of these are marked as `REQUIRED`. The system versions on Darwin are the preferred ones and will be given preference and if the dependencies are not found, the build would not be blocked. Of course, that is all Darwin specific.

Reply:
Oh right, of course. I'd forgotten these were not required by default